includes/wc-admin-order-functions.php

```diff
@@ -16,40 +16,80 @@ function wc_admin_order_product_lookup_entry( $order_id ) {
 	global $wpdb;
 
 	$order = wc_get_order( $order_id );
-	if ( ! $order ) {
+	if ( ! $order || 'shop_order_refund' === $order->get_type() ) {
 		return;
 	}
 
-	foreach ( $order->get_items() as $order_item ) {
-		$wpdb->replace(
+	if ( 'refunded' === $order->get_status() ) {
+		$wpdb->delete(
 			$wpdb->prefix . 'wc_order_product_lookup',
-			array(
-				'order_item_id'         => $order_item->get_id(),
-				'order_id'              => $order->get_id(),
-				'product_id'            => $order_item->get_product_id( 'edit' ),
-				'variation_id'          => $order_item->get_variation_id( 'edit' ),
-				'customer_id'           => ( 0 < $order->get_customer_id( 'edit' ) ) ? $order->get_customer_id( 'edit' ) : null,
-				'product_qty'           => $order_item->get_quantity( 'edit' ),
-				'product_gross_revenue' => $order_item->get_subtotal( 'edit' ),
-				'date_created'          => date( 'Y-m-d H:i:s', $order->get_date_created( 'edit' )->getTimestamp() ),
-			),
-			array(
-				'%d',
-				'%d',
-				'%d',
-				'%d',
-				'%d',
-				'%d',
-				'%f',
-				'%s',
-			)
+			array( 'order_id' => $order->get_id() )
```

Comment:
Do you want `array( '%d' )` as a third arg here?

Reply:
Added!